src/services/userService.ts

```diff
@@ -56,36 +56,72 @@ export default class UserService {
         return this.db.getCollection(USER_COLLECTION) as Observable<User[]>;
     }
 
-    updateUser(id: string, update: Object): Observable<void> {
-        return this.db.updateDoc(USER_COLLECTION, id, update);
+    updateUser(id: string, update: Object): Observable<User> {
+        return this.docRefObservableToUserObservable(this.db.updateDoc(USER_COLLECTION, id, update));
     }
 
-    updateUsers(users: User[]): Observable<firebase.firestore.DocumentReference[]> {
-        return this.db.updateDocs(USER_COLLECTION, users);
+    updateUsers(users: User[]): Observable<User[]> {
+        return this.docRefsObservableToUsersObservable(this.db.updateDocs(USER_COLLECTION, users));
     }
 
-    approveUser(id: string): Observable<void> {
-        return this.db.updateDoc(USER_COLLECTION, id, {
+    approveUser(id: string): Observable<User> {
+        return this.docRefObservableToUserObservable(this.db.updateDoc(USER_COLLECTION, id, {
             isApproved: true
-        });
+        }));
     }
 
-    disapproveUser(id: string): Observable<void> {
-        return this.db.updateDoc(USER_COLLECTION, id, {
+    approveUsers(ids: string[]): Observable<User[]> {
```

Comment:
we can simplify a lot of this code by passing in a bool value of `isAdmin` or `isApproved` that way we only need one method and it simplifies the front end code a lot

```
updateApproved(id: string, isApproved: boolean): Observable<User> { ... }
```